notebooks/pokemon_classifier.ipynb

```diff
@@ -323,18 +323,9 @@
     "    pokemon = pokemon[[\"pokedex_number\", \"name\", \"type1\", \"type2\"]]\n",
     "\n",
     "    # Create one-hot columns for each type\n",
-    "    types = set(pokemon[\"type1\"])\n",
-    "    for t in types:\n",
-    "        pokemon[\"is\" + str(t).capitalize()] = 0\n",
-    "\n",
-    "    # Iterate over Pokémon\n",
-    "    for i, p in pokemon.iterrows():\n",
-    "\n",
-    "        #  Set one-hot columns to 1 for relevant types\n",
-    "        pokemon.loc[i, \"is\" + p[\"type1\"].capitalize()] = 1\n",
-    "\n",
-    "        if not pd.isna(p[\"type2\"]):\n",
-    "            pokemon.loc[i, \"is\" + p[\"type2\"].capitalize()] = 1\n",
+    "    df_one_hot = pd.get_dummies(pokemon.type1.str.capitalize(), prefix='is', prefix_sep='')+pd.get_dummies(pokemon.type2.str.capitalize(), prefix='is', prefix_sep='')\n",
```

Comment:
In general, it is a good practice not to iterate over rows in Pandas as it is very inefficient. There is `pd.get_dummies` function that does the one-hot encoding.